src/logic/safe/utils/__tests__/safeVersion.test.ts

```diff
@@ -36,4 +40,70 @@ describe('Check safe version', () => {
+
+      jest
+        .spyOn(safeContracts, 'getMasterCopyAddressFromProxyAddress')
+        .mockImplementation(() => getSafeSingletonDeployment()?.networkAddresses['1'])
+      const isValid = await isValidMasterCopy('1', '0x0000000000000000000000000000000000000001', '1.3.0')
+      expect(isValid).toBe(true)
+    })
+
+    it('returns false for L1 mastercopy in L2 chain and version >=1.3.0', async () => {
+      jest.spyOn(config, '_getChainId').mockImplementation(() => '100')
+      jest.spyOn(config, 'getChainById').mockImplementation(() => ({ ...emptyChainInfo, chainId: '100', l2: true }))
+      jest
+        .spyOn(safeContracts, 'getMasterCopyAddressFromProxyAddress')
+        .mockImplementation(() => getSafeSingletonDeployment()?.networkAddresses['100'])
+      const isValid = await isValidMasterCopy('100', '0x0000000000000000000000000000000000000001', '1.3.0')
+      expect(isValid).toBe(false)
+    })
+
+    it('returns false for L1 mastercopy in L2 chain and version contains meta info', async () => {
+      jest.spyOn(config, '_getChainId').mockImplementation(() => '100')
+      jest.spyOn(config, 'getChainById').mockImplementation(() => ({ ...emptyChainInfo, chainId: '100', l2: true }))
+      jest
+        .spyOn(safeContracts, 'getMasterCopyAddressFromProxyAddress')
+        .mockImplementation(() => getSafeL2SingletonDeployment({ version: '1.1.1' })?.networkAddresses['100'])
+
+      const isValid = await isValidMasterCopy('100', '0x0000000000000000000000000000000000000001', '1.3.0+L2')
+      expect(isValid).toBe(false)
+    })
+
+    it('returns true for L1 mastercopy in L2 chain and version 1.1.1', async () => {
+      jest.spyOn(config, '_getChainId').mockImplementation(() => '100')
+      jest.spyOn(config, 'getChainById').mockImplementation(() => ({ ...emptyChainInfo, chainId: '100', l2: true }))
+      jest
+        .spyOn(safeContracts, 'getMasterCopyAddressFromProxyAddress')
+        .mockImplementation(() => getSafeSingletonDeployment({ version: '1.1.1' })?.networkAddresses['100'])
+      const isValid = await isValidMasterCopy('100', '0x0000000000000000000000000000000000000001', '1.1.1')
+      expect(isValid).toBe(true)
+    })
+
+    it('returns true for L1 mastercopy in L2 chain and version 1.2.0', async () => {
+      jest.spyOn(config, '_getChainId').mockImplementation(() => '100')
+      jest.spyOn(config, 'getChainById').mockImplementation(() => ({ ...emptyChainInfo, chainId: '100', l2: true }))
+      jest
+        .spyOn(safeContracts, 'getMasterCopyAddressFromProxyAddress')
+        .mockImplementation(() => getSafeSingletonDeployment({ version: '1.2.0' })?.networkAddresses['100'])
+      const isValid = await isValidMasterCopy('100', '0x0000000000000000000000000000000000000001', '1.2.0')
+      expect(isValid).toBe(true)
+    })
+
+    it('returns true for L1 mastercopy in L2 chain and version 1.0.0', async () => {
```

Comment:
What about an L2 version on the L1 chain?

Also a more general note: if a specific master copy is supported depends on the service running it. E.g. mainnet and GC support the all master copies while all other networks only support the 1.3.0+L2 version. Not sure what the expected behavior is for this PR, but if the goal is to show what version can be used with our services, we should probably rely on server side information.

Reply:
Is there an endpoint to get the server side information which versions are supported by our services for a chainID?
I agree that it would be the cleanest solution.

Reply:
I believe it's this one: https://safe-transaction.optimism.gnosis.io/api/v1/about/master-copies/, but it needs to be extended to flag non-indexed singletons

Reply:
There's a gateway endpoint too: `/v1/chains/{chainId}/about/master-copies`. The SDK has [master copy function](https://github.com/safe-global/safe-react-gateway-sdk/blob/main/src/index.ts#L247). It too would also need to be extended.

Reply:
Hmm could we probably detect it when looking at the `lastIndexedBlockNumber`?

For some L2 networks where the L1 master copy won't work like polygon it seems to stay the same as the `deployedBlockNumber`: https://safe-transaction.polygon.gnosis.io/api/v1/about/master-copies/

Reply:
Apparently the infra team was considering to remove all non-relevant / unsupported master copies and will do that soon.
So I will change the PR now to use this endpoint and simply check if the master copy version is included in the response.